subsys/net/lib/dns/dns_sd.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef DNS_SD_H_
+#define DNS_SD_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <net/dns_sd.h>
+#include <net/net_ip.h>
+
+#include "dns_pack.h"
+
+/* TODO: Move these into Kconfig */
+#define DNS_SD_PTR_TTL 4500
+#define DNS_SD_TXT_TTL 4500
+#define DNS_SD_SRV_TTL 120
+#define DNS_SD_A_TTL 120
+#define DNS_SD_AAAA_TTL 120
+
+#define DNS_SD_PTR_MASK (NS_CMPRSFLGS << 8)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define DNS_SD_FOREACH(it) \
+	Z_STRUCT_SECTION_FOREACH(dns_sd_rec, it)
+
+struct dns_header {
```

Comment:
Hmm, this struct might be useful to other work than just SD. Could we concentrate the generic symbols / structs to `dns_pack.h` for example.